_posts/2020-01-10-ios-logout-button-at-last.markdown

```diff
@@ -0,0 +1,93 @@
+want to encourage users to do.
+
+So Artsy staff could log out of the app, but our normal users couldn't. This quirk was acceptable in the early days
+of our app, but as the years wore on, it became less of a quirk and more of a product limitation. Even three months
+after creating our new Mobile Experience team, we hadn't yet prioritized this feature.
+
+Why? Well, to be honest, we had – and still have, [we're hiring by the way](http://artsy.net/jobs) – a lot of work
+to do to improve the app. Among the high-impact work we've been shipping, the log out button never made the cut.
+Why is that? Well, it's actually quite complicated. And why is it complicated? Well for that, we need to step back
+and think about software development and requirements gathering.
+
+Experienced software developers will tell you that it's far, far easier to build a piece of software with a feature
+in mind _from the start_ than it is to take an existing piece of software and add something to it that it was never
+intended to do. For our app, logging out was something it was never intended to do. So adding it was hard. Why?
+
+Well, because of state. When you log in to our app, the state of the app changes: we get a user ID and access token
+from our API, and our code assumed that these values would never change. This is further complicated by the fact
+that our app is split into two pieces: the native code (written in Objective-C and Swift) and the React Native code
+(written in TypeScript). You can effectively think of these as _two_ apps that interoperate with each other. The
+state now has to be managed across _two_ pieces of software, further complexifying the work to add a log out
+button.
+
+Returning to the idea of developing features in mind from the start (versus adding them after the fact), our React
+Native codebase had always assumed a logged-in user. Adding support for this later on was too difficult (indeed,
+our solution was to
+[invalidate the React Native runtime upon log out](https://github.com/artsy/emission/pull/2027/files#diff-0cc174f9197fd0b06ecbd2eaa0247833R1020)).
+This wasn't a limitation of React Native, but rather it was a limitation of how we chose to organize our code. If
+we'd added a log out button earlier in the product lifetime, it wouldn't have been so difficult. The longer we
+waited, the more and more code we wrote that implicitly relied upon our existing limitations.
+
+In this way, the _absence_ of a feature had gained its own inertia. The missing feature became a present absence,
+and I think there's more to think about there – maybe for another blog post.
+
+So what we had was a difficult technical problem that wasn't _that_ high of priority and didn't have an obvious
+solution. Our product team wasn't feeling the pain, but our colleagues who interface more directly with our users
+_were_ feeling it.
+
+Changing gears a bit, Artsy kicked off 2020 with a company-wide Hackathon. We run these events roughly once year,
```

Comment:
I think you could just cut `Changing gears a bit,` - it slows the momentum that you've built here